metricbeat/module/kubernetes/state_node/state_node_test.go

```diff
@@ -20,14 +20,36 @@
 package state_node
 
 import (
+	"encoding/json"
+	"flag"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"sort"
 	"testing"
 
-	"github.com/elastic/beats/metricbeat/helper/prometheus/ptest"
+	"github.com/mitchellh/hashstructure"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/elastic/beats/libbeat/common"
+	mbtest "github.com/elastic/beats/metricbeat/mb/testing"
 )
 
+// TestCases holds the list of test cases to test a metricset
+type TestCases []struct {
+	// MetricsFile containing Prometheus outputted metrics
+	MetricsFile string
+
+	// ExpectedFile containing resulting documents
+	ExpectedFile string
+}
+
+var expectedFlag = flag.Bool("update_expected", false, "Update prometheus expected files")
+
 func TestEventMapping(t *testing.T) {
-	ptest.TestMetricSetEventsFetcher(t, "kubernetes", "state_node",
-		ptest.TestCases{
+	testMetricSetEventsFetcher(t, "kubernetes", "state_node",
```

Comment:
I assume as soon as we have migrated to K8s metricsets to V2 this can be simplified again? Perhaps it can even be based on the new http test framework?

Reply:
In fact, there was a `ptest.TestMetricSet` that works with V2 and the code duplication is not needed anymore (but that function name was unfortunate). I have removed it already.

As a side note, I don't think it's a good idea to have such dependency with Prometheus.